WooCommerce/WooCommerceScreenshots/WooCommerceScreenshots.swift

```diff
@@ -54,6 +60,67 @@ class WooCommerceScreenshots: XCTestCase {
             .selectProduct(atIndex: 1)
             .thenTakeScreenshot(named: "product-details")
     }
+
+    private let port = Int.random(in: 2048...65535)
+
+    private let loop = try! SelectorEventLoop(selector: try! KqueueSelector())
+
+    private let queue = DispatchQueue(label: "screenshots-asset-server")
+
+    private lazy var server = DefaultHTTPServer(eventLoop: loop, port: port, app: handleWebRequest)
+
+    func startWebServer() {
+        queue.async {
+            /// Start HTTP server to listen on the port
+            try! self.server.start()
+
+            /// Run event loop
+            self.loop.runForever()
+
+            debugPrint("Web Server running on port \(self.port)")
+        }
+    }
+
+    func handleWebRequest(env: [String: Any], startResponse: ((String, [(String, String)]) -> ()), sendBody: ((Data) -> ())) {
+        /// Extract the path, so http://localhost:9285/foo-bar would be `/foo-bar` here.
+        guard let path = env["PATH_INFO"] as? String else {
```

Comment:
Always found it odd that Embassy used an untyped dictionary to access those values, as opposed to a type-checked structure, to expose those values; feels quite unnatural in Swift…

Maybe worth a comment to at least point to doc:
```suggestion
        // See https://github.com/envoy/Embassy#whats-swsgi-swift-web-server-gateway-interface for more about PATH_INFO
        guard let path = env["PATH_INFO"] as? String else {
```

Reply:
Yeah good call – addressed in 896a0404b8efdcf96458c9fb7bd7b8f42dee8f09 – I didn't use the suggestion so I could add a formatted comment which is the woo default